tests/test_integrity_checker.py

```diff
@@ -121,6 +121,33 @@ def test_variable_doesnt_exist(self):
 
 class IntegrityCheckerTestCase(TestCase):
 
+    def test_ignore_private_methods(self):
+        program = '\n'.join([
+            'def function_with_missing_parameter(x):',
+            '    """We\'re missing a description of x."""',
+            '    print(x / 2)',
+            ''
+            'def _same_error_but_private_method(x):',
+            '    """We\'re missing a description of x."""',
+            '    print(x / 2)',
+        ])
+        tree = ast.parse(program)
+        functions = get_function_descriptions(tree)
+        checker = IntegrityChecker(
+            config=Configuration(
+                ignore=[],
+                message_template=None,
+                style=DocstringStyle.GOOGLE,
+                strictness=Strictness.FULL_DESCRIPTION,
+                ignore_regex=r'^_(.*)'
+            )
+        )
+        checker.schedule(functions[0])
+        checker.schedule(functions[1])
```

Comment:
You should use `checker.run_checks(functions[0])` in tests.  `IntegrityChecker.schedule` adds `IntegrityChecker.run_checks` to a ThreadPoolExecutor (it's asynchronous).  This test will likely fail under some conditions because the test never shuts down the executor.

Probably, I should have added a docstring to `IntegrityChecker.schedule`.  (Making the function async would be even better, but I'd like to support earlier versions of Python3.)

Reply:
Hmm I already feared something like that. It is now the scheduler that decides whether a check should run or not (see the added `if`-clause on [integrity_checker.py:95](https://github.com/terrencepreilly/darglint/pull/57/files#diff-9d2303b11308c4a737d5eed53148b5f7R95)). I could modify this test and mock the executor or something? Or we could make the decision to skip a certain function for check in the `checker.run_checks` method, but I'm not sure if that would be the right place. For the existing check (skipping checks for functions that have no docstrings) the exclusion logic happens both in the `IntegrityChecker.schedule` as well as in the `IntegrityChecker.run_checks` method. Would like to hear what you think.

Reply:
Ah, I see what you mean.  Unfortunately, there's not a central place of control for these sort of things. (This probably needs to be refactored at some point.)

If we only wanted to add this `if-`clause to one place, then it should be in `IntegrityChecker.run_checks`, since that is ultimately the function which performs the checks, and so is guaranteed to be called at some point.  For example, the current code will not work with flake8, because the flake8 handler calls `IntegrityChecker.run_checks`.  (This is probably an oversight on my part -- I just searched for instances of `run_checks` and found this out.)

Feel free to add this conditional to both `IntegrityChecker.run_checks` and `IntegrityChecker.schedule`.  (But just call `IntegrityChecker.run_checks` in your test.)  This will constitute the [third repetition](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) between these two methods, so it will be a good excuse for me to refactor it.

Reply:
I updated this. I've created a new private method (`IntegrityChecker._skip_checks`) that is used both in the `run_checks` as well as the `schedule` methods. Also, test now use the `run_checks` method.

Reply:
Looks good! Thanks!